crev-lib/src/lib.rs

```diff
@@ -196,35 +196,49 @@ where
     ))
 }
 
+fn parse_url_and_username(git_url_or_github_username: &str) -> (String, Option<String>) {
+    let mut git_https_url;
+    let mut github_username;
+
+    let is_username = !git_url_or_github_username.contains('/');
+    if is_username {
+        github_username = Some(git_url_or_github_username.to_string());
+        git_https_url = format!("https://github.com/{}/crev-proofs", git_url_or_github_username);
+    } else {
+        git_https_url = git_url_or_github_username.to_string();
+        match self::local::parse_git_url_https(&git_https_url) {
+            Some(components) => {
+                github_username = Some(components.username);
+            },
+            None => {
+                github_username = None;
+            }
+        }
+    }
+
+    (git_https_url, github_username)
+}
+
 pub fn generate_id() -> Result<()> {
-    eprintln!("Enter URL of your Proof Repository to associate with the new CrevId");
-    eprintln!("E.g.: https://github.com/<yourusername>/crev-proofs");
+    eprintln!("Enter a public git URL or Github username for publishing your CrevID proofs");
+    eprintln!("Example: https://github.com/<myusername>/crev-proofs");
     eprintln!("or just your github username to generate it.");
     eprintln!("Visit https://github.com/dpc/crev/wiki/Proof-Repository for help.");
-    let mut url;
-    loop {
-        eprintln!("");
-        url = rprompt::prompt_reply_stdout("URL or Github username: ")?;
-        if !url.contains('/') {
-            url = format!("https://github.com/{}/crev-proofs", url)
-        }
-        eprintln!("");
-        eprintln!("Your URL: {}", url);
-        eprintln!("It is recomended that this repository exists and is initialized upfront (can be empty).");
-        if crev_common::yes_or_no_was_y("Proceed? (y/n) ")? {
-            break;
-        }
-    }
+
+    let git_url_or_github_username = rprompt::prompt_reply_stdout("Git URL or Github username: ")?;
+    eprintln!("");
+
+    let (git_https_url, github_username) = parse_url_and_username(&git_url_or_github_username);
+    eprintln!("Repository URL: {}\n", git_https_url);
+    eprintln!("It is recomended that this repository exists and is initialized upfront (can be empty).");
```

Comment:
Why no loop? What if user made a typo? :)

Please don't change this part. Just yesterday I went through  https://github.com/dpc/crev/issues/75 to address it. If you think it should be changed/improved, let's discuss it slowly in a separate issue and PR.

Reply:
I am actually going to scrap the whole thing and change it. There is going to be no interactive prompts or anything, except for asking for a pasword. So please revert this part.